src/core/server/saved_objects/import/import_saved_objects.ts

```diff
@@ -108,6 +110,20 @@ export async function importSavedObjectsFromStream({
   );
   errorAccumulator = [...errorAccumulator, ...validateReferencesResult];
 
+  if (isCopy) {
+    // Data sources can only be assigned to workspaces and can not be copied between workspaces.
+    collectSavedObjectsResult.collectedObjects = collectSavedObjectsResult.collectedObjects.filter(
+      (obj) => obj.type !== 'data-source'
+    );
+    const validateDataSourcesResult = await validateDataSources(
```

Comment:
Shall we add a check here to see if data source is enabled? Or I guess the await here will throw error.

Reply:
If the data source is not enabled, the objects of `validateDataSourcesResult` will contain the missing data source error and not be copied to the target workspace.

Reply:
If data source is not enabled, there is no need to `validateDataSources` right? Luckily the find method in repository won't throw error for unsupported type but get will. I just feel it more safe and clean that we only validate when data source is enabled.

Reply:
No, even if the data source is not enabled, we also need to `validateDataSources`. Because if objects belong to a certain datasource before closing the datasource, they are not allowed to be copied.  If there is no verification, these objects cannot be used even if they are copied.

Reply:
The validation makes additional nested bulkGet calls, so doing the validation under some feature flag would be ideal. But anyway, `isCopy` will only be true when workspace is enabled. I am OK with that.